private-templates-service/client/src/App.tsx

```diff
@@ -0,0 +1,212 @@
+          debug={this.state.error.debug}
+        />
+      );
+    }
+
+    return (
+      <Router>
+        <div>
+          <NavBar
+            isAuthenticated={this.state.isAuthenticated}
+            authButtonMethod={
+              this.state.isAuthenticated
+                ? this.logout
+                : this.login
+            }
+            user={this.state.user}
+          />
+          <Container>
+            {error}
+            <Route
+              exact
+              path="/"
+              render={props => (
+                <Welcome
+                  {...props}
+                  isAuthenticated={this.state.isAuthenticated}
+                  user={this.state.user}
+                  authButtonMethod={this.login}
+                />
+              )}
+            />
+          </Container>
+        </div>
+      </Router>
+    );
+  }
+
+  login = async () => {
+    try {
+      await this.userAgentApplication.loginPopup({
+        scopes: config.scopes,
+        prompt: "select_account"
+      });
+      await this.getUserProfile();
+    } catch (err) {
+      var error = {};
+
+      switch (true) {
+        case (typeof err === "string"):
+          var errParts = err.split("|");
+          error =
+            errParts.length > 1
+              ? {
+                message: errParts[1],
+                debug: errParts[0]
+              }
+              : {
+                message: err
+              };
+          break;
+        case (err instanceof ClientAuthError):
+          error = {
+            message: err.message,
+            debug: JSON.stringify(err)
+          };
+          break;
+        default:
+          console.log("Unexpected authentication error: ", err);
+      }
+
+      this.setState({
+        isAuthenticated: false,
+        user: {},
+        error: error
+      });
+    }
+  }
+
+  logout = () => {
+    this.userAgentApplication.logout();
+  }
+
+  getUserProfile = async () => {
+    try {
+      // Get the access token silently
+      // If the cache contains a non-expired token, this function
+      // will just return the cached token. Otherwise, it will
+      // make a request to the Azure OAuth endpoint to get a token
+
+      var accessToken = await this.userAgentApplication.acquireTokenSilent(
+        {
+          scopes: config.scopes
+        }
+      );
+
+      if (accessToken) {
+        // Get the user's profile from Graph
+        var user = await getUserDetails(accessToken);
+        var org = await getOrgDetails(accessToken);
+        if (org.value.length === 0) {
+          org = null;
+        } else {
+          org = org.value[0].displayName;
+        }
+        this.setState({
+          isAuthenticated: true,
+          user: {
+            displayName: user.displayName,
+            email: user.mail || user.userPrincipalName,
+            organization: org
+          },
+          error: null
+        });
+      }
+    } catch (err) {
+      var error = {};
```

Comment:
If we type this from the start, it will ensure the code doesn't add extra props that aren't valid and also has any required ones (if you update the ErrorMessageProps type later for example, the compilation will break here - which is what you want).

Reply:
`var error: ErrorMessageProps;`